edmq_backend/src/scripts/shazam_data.py

```diff
@@ -0,0 +1,22 @@
+import requests
+import json
+
+# Monthly quota is 500 from RapidAPI
+shazam_headers = {
+    'x-rapidapi-key': "194d3fca7emsh5f18f2dac01d040p176829jsnf6975cc79831",
+    'x-rapidapi-host': "shazam.p.rapidapi.com"
+    }
+shazam_url = "https://shazam.p.rapidapi.com/search"
+
+def get_title_artist(term):
+    d = {}
+    querystring = {"term": term, "locale": "en-US", "offset": "0","limit": "1"}
+    search_response = requests.request("GET", shazam_url, headers=shazam_headers, params=querystring).text
+    search_dict = json.loads(search_response)
```

Comment:
can you provide a comment describing what this json looks like

Reply:
{
   "tracks":{
      "hits":[
         {
            "track":{
               "layout":"5",
               "type":"MUSIC",
               "key":"332171602",
               "title":"Rush Over Me",
               "subtitle":"Seven Lions & Illenium & Said The Sky Feat. HALIENE",
               "share":{
                  "subject":"Rush Over Me - Seven Lions & Illenium & Said The Sky Feat. HALIENE",
                  "text":"I used Shazam to discover Rush Over Me by Seven Lions & Illenium & Said The Sky Feat. HALIENE.",
                  "href":"https://www.shazam.com/track/332171602/rush-over-me",
                  "image":"https://images.shazam.com/coverart/t332171602-i1321272674_s400.jpg",
                  "twitter":"I used @Shazam to discover Rush Over Me by Seven Lions & Illenium & Said The Sky Feat. HALIENE.",
                  "html":"https://www.shazam.com/snippets/email-share/332171602?lang=en-US&country=US",
                  "avatar":"https://is5-ssl.mzstatic.com/image/thumb/Features124/v4/2d/87/e2/2d87e2bb-d3f9-7a3b-9846-41ec138de363/pr_source.png/800x800bb.jpeg",
                  "snapchat":"https://www.shazam.com/partner/sc/track/332171602"
               },
               "images":{
                  "background":"https://is5-ssl.mzstatic.com/image/thumb/Features124/v4/2d/87/e2/2d87e2bb-d3f9-7a3b-9846-41ec138de363/pr_source.png/800x800bb.jpeg",
                  "coverart":"https://images.shazam.com/coverart/t332171602-b1321272674_s400.jpg",
                  "coverarthq":"https://images.shazam.com/coverart/t332171602-b1321272674_s800.jpg"
               },
               "hub":{
                  "type":"APPLEMUSIC",
                  "image":"https://images.shazam.com/static/icons/hub/ios/v5/applemusic_{scalefactor}.png",
                  "actions":[
                     {
                        "name":"apple",
                        "type":"applemusicplay",
                        "id":"1321272674"
                     },
                     {
                        "name":"apple",
                        "type":"uri",
                        "uri":"https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview128/v4/e1/fb/34/e1fb344b-16e8-b714-e0ca-fba1ea3b2c09/mzaf_4864775246187120862.plus.aac.p.m4a"
                     }
                  ],
                  "options":[
                     {
                        "caption":"OPEN",
                        "actions":[
                           {
                              "name":"hub:applemusic:deeplink",
                              "type":"applemusicopen",
                              "uri":"https://music.apple.com/us/album/rush-over-me-feat-haliene/1321272511?i=1321272674&uo=5&at=1001l4DI&ct=5348615A-616D-3235-3830-44754D6D5973&mttnagencyid=769459046716559743&mttnsiteid=125115&mttn3pid=a_custom_779816081798873874&mttnsub1=Shazam_ios&mttnsub2=5348615A-616D-3235-3830-44754D6D5973&itscg=30201&app=music&itsct=Shazam_ios"
                           },
                           {
                              "name":"hub:applemusic:deeplink",
                              "type":"uri",
                              "uri":"https://music.apple.com/us/album/rush-over-me-feat-haliene/1321272511?i=1321272674&uo=5&at=1001l4DI&ct=5348615A-616D-3235-3830-44754D6D5973&mttnagencyid=769459046716559743&mttnsiteid=125115&mttn3pid=a_custom_779816081798873874&mttnsub1=Shazam_ios&mttnsub2=5348615A-616D-3235-3830-44754D6D5973&itscg=30201&app=music&itsct=Shazam_ios"
                           }
                        ],
                        "beacondata":{
                           "type":"open",
                           "providername":"applemusic"
                        },
                        "image":"https://images.shazam.com/static/icons/hub/ios/v5/overflow-open-option_{scalefactor}.png",
                        "type":"open",
                        "listcaption":"Open in Apple Music",
                        "overflowimage":"https://images.shazam.com/static/icons/hub/ios/v5/applemusic-overflow_{scalefactor}.png",
                        "colouroverflowimage":false,
                        "providername":"applemusic"
                     },
                     {
                        "caption":"BUY",
                        "actions":[
                           {
                              "type":"itunesbuy",
                              "uri":"https://itunes.apple.com/us/album/rush-over-me-feat-haliene/1321272511?i=1321272674&uo=5&at=11l3eE&ct=5348615A-616D-3235-3830-44754D6D5973&app=itunes"
                           }
                        ],
                        "beacondata":{
                           "type":"buy",
                           "providername":"itunes"
                        },
                        "image":"https://images.shazam.com/static/icons/hub/ios/v5/itunes-overflow-buy_{scalefactor}.png",
                        "type":"buy",
                        "listcaption":"Buy on iTunes",
                        "overflowimage":"https://images.shazam.com/static/icons/hub/ios/v5/itunes-overflow-buy_{scalefactor}.png",
                        "colouroverflowimage":false,
                        "providername":"itunes"
                     }
                  ],
                  "providers":[
                     {
                        "caption":"Open in Spotify",
                        "images":{
                           "overflow":"https://images.shazam.com/static/icons/hub/ios/v5/spotify-overflow_{scalefactor}.png",
                           "default":"https://images.shazam.com/static/icons/hub/ios/v5/spotify_{scalefactor}.png"
                        },
                        "actions":[
                           {
                              "name":"hub:spotify:androidsearchdeeplink",
                              "type":"intent",
                              "uri":"intent:#Intent;action=android.media.action.MEDIA_PLAY_FROM_SEARCH;component=com.spotify.music/.MainActivity;S.referrer=Shazam;S.query=Rush%20Over%20Me%20Seven%20Lions;end"
                           },
                           {
                              "name":"hub:spotify:iossearchdeeplink",
                              "type":"uri",
                              "uri":"spotify:search:Rush%20Over%20Me%20Seven%20Lions"
                           }
                        ],
                        "type":"SPOTIFY"
                     },
                     {
                        "caption":"Open in Deezer",
                        "images":{
                           "overflow":"https://images.shazam.com/static/icons/hub/ios/v5/deezer-overflow_{scalefactor}.png",
                           "default":"https://images.shazam.com/static/icons/hub/ios/v5/deezer_{scalefactor}.png"
                        },
                        "actions":[
                           {
                              "name":"hub:deezer:searchdeeplink",
                              "type":"uri",
                              "uri":"deezer-query://www.deezer.com/play?query=%7Btrack%3A%27Rush+Over+Me%27%20artist%3A%27Seven+Lions++Illenium++Said+The+Sky%27%7D"
                           }
                        ],
                        "type":"DEEZER"
                     }
                  ],
                  "explicit":false,
                  "displayname":"APPLE MUSIC"
               },
               "artists":[
                  {
                     "id":"41223759",
                     "adamid":"461065242"
                  }
               ],
               "url":"https://www.shazam.com/track/332171602/rush-over-me"
            }
         }
      ]
   },
   "artists":{
      "hits":[
         {
            "artist":{
               "avatar":"https://is1-ssl.mzstatic.com/image/thumb/Music113/v4/b4/78/62/b4786242-9399-a4be-a2d9-815311bc2798/pr_source.png/800x800bb.jpeg",
               "id":"201821084",
               "name":"HALIENE",
               "verified":false
            }
         }
      ]
   }
}

Reply:
{"tracks":{"hits":[{"track":{"layout":"5","type":"MUSIC","key":"332171602","title":"Rush Over Me","subtitle":"Seven Lions & Illenium & Said The Sky Feat. HALIENE","share":{"subject":"Rush Over Me - Seven Lions & Illenium & Said The Sky Feat. HALIENE","text":"I used Shazam to discover Rush Over Me by Seven Lions & Illenium & Said The Sky Feat. HALIENE.","href":"https://www.shazam.com/track/332171602/rush-over-me","image":"https://images.shazam.com/coverart/t332171602-i1321272674_s400.jpg","twitter":"I used @Shazam to discover Rush Over Me by Seven Lions & Illenium & Said The Sky Feat. HALIENE.","html":"https://www.shazam.com/snippets/email-share/332171602?lang=en-US&country=US","avatar":"https://is5-ssl.mzstatic.com/image/thumb/Features124/v4/2d/87/e2/2d87e2bb-d3f9-7a3b-9846-41ec138de363/pr_source.png/800x800bb.jpeg","snapchat":"https://www.shazam.com/partner/sc/track/332171602"},"images":{"background":"https://is5-ssl.mzstatic.com/image/thumb/Features124/v4/2d/87/e2/2d87e2bb-d3f9-7a3b-9846-41ec138de363/pr_source.png/800x800bb.jpeg","coverart":"https://images.shazam.com/coverart/t332171602-b1321272674_s400.jpg","coverarthq":"https://images.shazam.com/coverart/t332171602-b1321272674_s800.jpg"},"hub":{"type":"APPLEMUSIC","image":"https://images.shazam.com/static/icons/hub/ios/v5/applemusic_{scalefactor}.png","actions":[{"name":"apple","type":"applemusicplay","id":"1321272674"},{"name":"apple","type":"uri","uri":"https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview128/v4/e1/fb/34/e1fb344b-16e8-b714-e0ca-fba1ea3b2c09/mzaf_4864775246187120862.plus.aac.p.m4a"}],"options":[{"caption":"OPEN","actions":[{"name":"hub:applemusic:deeplink","type":"applemusicopen","uri":"https://music.apple.com/us/album/rush-over-me-feat-haliene/1321272511?i=1321272674&uo=5&at=1001l4DI&ct=5348615A-616D-3235-3830-44754D6D5973&mttnagencyid=769459046716559743&mttnsiteid=125115&mttn3pid=a_custom_779816081798873874&mttnsub1=Shazam_ios&mttnsub2=5348615A-616D-3235-3830-44754D6D5973&itscg=30201&app=music&itsct=Shazam_ios"},{"name":"hub:applemusic:deeplink","type":"uri","uri":"https://music.apple.com/us/album/rush-over-me-feat-haliene/1321272511?i=1321272674&uo=5&at=1001l4DI&ct=5348615A-616D-3235-3830-44754D6D5973&mttnagencyid=769459046716559743&mttnsiteid=125115&mttn3pid=a_custom_779816081798873874&mttnsub1=Shazam_ios&mttnsub2=5348615A-616D-3235-3830-44754D6D5973&itscg=30201&app=music&itsct=Shazam_ios"}],"beacondata":{"type":"open","providername":"applemusic"},"image":"https://images.shazam.com/static/icons/hub/ios/v5/overflow-open-option_{scalefactor}.png","type":"open","listcaption":"Open in Apple Music","overflowimage":"https://images.shazam.com/static/icons/hub/ios/v5/applemusic-overflow_{scalefactor}.png","colouroverflowimage":false,"providername":"applemusic"},{"caption":"BUY","actions":[{"type":"itunesbuy","uri":"https://itunes.apple.com/us/album/rush-over-me-feat-haliene/1321272511?i=1321272674&uo=5&at=11l3eE&ct=5348615A-616D-3235-3830-44754D6D5973&app=itunes"}],"beacondata":{"type":"buy","providername":"itunes"},"image":"https://images.shazam.com/static/icons/hub/ios/v5/itunes-overflow-buy_{scalefactor}.png","type":"buy","listcaption":"Buy on iTunes","overflowimage":"https://images.shazam.com/static/icons/hub/ios/v5/itunes-overflow-buy_{scalefactor}.png","colouroverflowimage":false,"providername":"itunes"}],"providers":[{"caption":"Open in Spotify","images":{"overflow":"https://images.shazam.com/static/icons/hub/ios/v5/spotify-overflow_{scalefactor}.png","default":"https://images.shazam.com/static/icons/hub/ios/v5/spotify_{scalefactor}.png"},"actions":[{"name":"hub:spotify:androidsearchdeeplink","type":"intent","uri":"intent:#Intent;action=android.media.action.MEDIA_PLAY_FROM_SEARCH;component=com.spotify.music/.MainActivity;S.referrer=Shazam;S.query=Rush%20Over%20Me%20Seven%20Lions;end"},{"name":"hub:spotify:iossearchdeeplink","type":"uri","uri":"spotify:search:Rush%20Over%20Me%20Seven%20Lions"}],"type":"SPOTIFY"},{"caption":"Open in Deezer","images":{"overflow":"https://images.shazam.com/static/icons/hub/ios/v5/deezer-overflow_{scalefactor}.png","default":"https://images.shazam.com/static/icons/hub/ios/v5/deezer_{scalefactor}.png"},"actions":[{"name":"hub:deezer:searchdeeplink","type":"uri","uri":"deezer-query://www.deezer.com/play?query=%7Btrack%3A%27Rush+Over+Me%27%20artist%3A%27Seven+Lions++Illenium++Said+The+Sky%27%7D"}],"type":"DEEZER"}],"explicit":false,"displayname":"APPLE MUSIC"},"artists":[{"id":"41223759","adamid":"461065242"}],"url":"https://www.shazam.com/track/332171602/rush-over-me"}}]},"artists":{"hits":[{"artist":{"avatar":"https://is1-ssl.mzstatic.com/image/thumb/Music113/v4/b4/78/62/b4786242-9399-a4be-a2d9-815311bc2798/pr_source.png/800x800bb.jpeg","id":"201821084","name":"HALIENE","verified":false}}]}}